tests/property/ic_api/cycles/src/intermediary/index.ts

```diff
@@ -0,0 +1,52 @@
+import { call, IDL, msgCyclesRefunded, trap, update } from 'azle';
+
+import { CyclesResult, CyclesResultIDL } from '../types';
+
+export default class {
+    @update([IDL.Nat64], CyclesResultIDL)
+    async sendAllCycles(amount: bigint): Promise<CyclesResult> {
+        const result = await call(getCyclesPrincipal(), 'receiveAllCycles', {
+            returnIdlType: CyclesResultIDL,
+            cycles: amount
+        });
+        return { ...result, cyclesRefunded: msgCyclesRefunded() };
+    }
+
+    @update([IDL.Nat64], CyclesResultIDL)
+    async sendHalfCycles(amount: bigint): Promise<CyclesResult> {
+        const result = await call(getCyclesPrincipal(), 'receiveHalfCycles', {
+            returnIdlType: CyclesResultIDL,
+            cycles: amount
+        });
+        return { ...result, cyclesRefunded: msgCyclesRefunded() };
+    }
+
+    @update([IDL.Nat64], CyclesResultIDL)
+    async sendNoCycles(amount: bigint): Promise<CyclesResult> {
+        const result = await call(getCyclesPrincipal(), 'receiveNoCycles', {
+            returnIdlType: CyclesResultIDL,
+            cycles: amount
+        });
+        return { ...result, cyclesRefunded: msgCyclesRefunded() };
+    }
+
+    @update([IDL.Nat64], IDL.Nat64)
+    async sendCyclesManual(amount: bigint): Promise<bigint> {
+        console.log(
+            'sendCyclesManual And its for sure this unquie one from 1:17',
+            amount
+        );
+        const result = await call(getCyclesPrincipal(), 'receiveCyclesManual', {
+            returnIdlType: IDL.Nat64,
+            cycles: amount
+        });
+        return result;
+    }
+}
+
+function getCyclesPrincipal(): string {
```

Comment:
Have a class instance variable initialized by calling this function

Reply:
done